onnxruntime/core/optimizer/gelu_fusion.cc

```diff
@@ -0,0 +1,204 @@
+      continue;
+    }
+
+    const Node& mul_node = *(add_node.OutputNodesBegin());
+    if (!graph_utils::IsSupportedOptypeVersionAndDomain(mul_node, "Mul", {7}) ||
+        mul_node.GetExecutionProviderType() != div.GetExecutionProviderType() ||
+        !IsSupportedDataType(mul_node)) {
+      continue;
+    }
+
+    const Node* mul2_node = nullptr;
+    for (auto iter = mul_node.InputNodesBegin(); iter != mul_node.InputNodesEnd(); ++iter) {
+      if ((*iter).OpType().compare("Mul") == 0) {
+        // find the other input node of Mul
+        mul2_node = &(*iter);
+        break;
+      }
+    }
+    if (mul2_node == nullptr) {
+      continue;
+    }
+
+    if (!graph_utils::IsSupportedOptypeVersionAndDomain(*mul2_node, "Mul", {7}) ||
+        mul2_node->GetExecutionProviderType() != div.GetExecutionProviderType() ||
+        mul2_node->GetOutputEdgesCount() != 1 ||
+        !IsSupportedDataType(*mul2_node)) {
+      continue;
+    }
+
+    // Check the other input node(e.g. not of type Add) is 0.5f.
+    int mul_const_input_index = 0;
+    if (mul2_node->InputDefs()[0]->Name() == div.MutableInputDefs()[0]->Name()) {
+      mul_const_input_index = 1;
+    }
+
+    const auto& mul_const_input_arg = mul2_node->InputDefs()[mul_const_input_index];
+    if (!CheckConstantInput(graph, *mul_const_input_arg, 0.5f)) {
+      continue;
+    }
+
+    const std::vector<NodeArg*> gelu_input_defs{div.MutableInputDefs()[0]};
+    const std::vector<NodeArg*> gelu_output_defs{const_cast<NodeArg*>(mul_node.OutputDefs()[0])};
+    Node& gelu_node = graph.AddNode(graph.GenerateNodeName("Gelu"),
+                                    "Gelu",
+                                    "fused Gelu subgraphs ",
+                                    gelu_input_defs,
+                                    gelu_output_defs, {}, kMSDomain);
+
+    // Assign provider to this new node. Provider should be same as the provider for old node.
+    gelu_node.SetExecutionProviderType(div.GetExecutionProviderType());
+
+    removed_nodes.push_front(div.Index());
+    removed_nodes.push_front(erf_node.Index());
+    removed_nodes.push_front(add_node.Index());
+    removed_nodes.push_front(mul2_node->Index());
+    removed_nodes.push_front(mul_node.Index());
+
+    // The initializers will be candidiates for removal.
+    // While we will check their usage before removing them from graph.
+    removed_initializers.push_front(div.MutableInputDefs()[1]);
+    removed_initializers.push_front(const_cast<NodeArg*>(add_const_input_arg));
+    removed_initializers.push_front(const_cast<NodeArg*>(mul_const_input_arg));
+  }
+
+  // Have to remove node in reversed order for now to walk around the issue in RemoveNode
+  for (onnxruntime::NodeIndex removed_node : removed_nodes) {
+    graph.RemoveNode(removed_node);
+  }
+
+  for (auto& tensor_arg : removed_initializers) {
+    if (!HasConsumers(graph, tensor_arg)) {
+      graph.RemoveInitializedTensor(tensor_arg->Name());
+    }
+  }
```

Comment:
Graph::Resolve will handle this so no need to do it here.

Reply:
Yeah, good point.